### ISSUE TITLE
ImplicitUnitReturnType false positive

### ISSUE THREAD

**Issue by marschwar** [detekt/detekt#4004]
## Expected Behavior

```kotlin
fun foo() = Unit
```

should not violate the `ImplicitUnitReturnType`rule as there is nothing implicit about the return type if the function body is simply `Unit`. What makes matters worse is the fact that the `OptionalUnit` rule explicitly suggests `override fun foo() = Unit`.
